src/scaffoldmaker/utils/zinc_utils.py

```diff
@@ -354,33 +389,32 @@ def exnodeStringFromNodeValues(
     context = Context('exnodeStringFromNodeValues')
     region = context.getDefaultRegion()
-    fm = region.getFieldmodule()
-    fm.beginChange()
-    cache = fm.createFieldcache()
-    coordinates = getOrCreateCoordinateField(fm, componentsCount = componentsCount)
-    nodes = fm.findNodesetByFieldDomainType(Field.DOMAIN_TYPE_NODES)
-    group = fm.createFieldGroup()
-    group.setName(groupName)
-    nodesetGroup = group.createFieldNodeGroup(nodes).getNodesetGroup()
-    nodetemplate = nodes.createNodetemplate()
-    nodetemplate.defineField(coordinates)
-    if not Node.VALUE_LABEL_VALUE in nodeValueLabels:
-        nodetemplate.setValueNumberOfVersions(coordinates, -1, Node.VALUE_LABEL_VALUE, 0)
-    for nodeValueLabel in nodeValueLabels:
-        nodetemplate.setValueNumberOfVersions(coordinates, -1, nodeValueLabel, 1)
-    # create nodes
-    for n in range(nodesCount):
-        node = nodesetGroup.createNode(n + 1, nodetemplate)
-        cache.setNode(node)
-        for v in range(nodeValueLabelsCount):
-            coordinates.setNodeParameters(cache, -1, nodeValueLabels[v], 1, nodeValues[n][v])
-    # serialise to string
-    sir = region.createStreaminformationRegion()
-    srm = sir.createStreamresourceMemory()
-    sir.setResourceGroupName(srm, groupName)
-    region.write(sir)
-    result, exString = srm.getBuffer()
-    #print('\n',nodeValues)
-    #print('exnodeStringFromNodeValues', result, exString)
-    fm.endChange()
+    fieldmodule = region.getFieldmodule()
+    with ZincCacheChanges(fieldmodule):
+        cache = fieldmodule.createFieldcache()
+        coordinates = getOrCreateCoordinateField(fieldmodule, componentsCount = componentsCount)
+        nodes = fieldmodule.findNodesetByFieldDomainType(Field.DOMAIN_TYPE_NODES)
+        group = fieldmodule.createFieldGroup()
+        group.setName(groupName)
+        nodesetGroup = group.createFieldNodeGroup(nodes).getNodesetGroup()
+        nodetemplate = nodes.createNodetemplate()
+        nodetemplate.defineField(coordinates)
+        if not Node.VALUE_LABEL_VALUE in nodeValueLabels:
+            nodetemplate.setValueNumberOfVersions(coordinates, -1, Node.VALUE_LABEL_VALUE, 0)
+        for nodeValueLabel in nodeValueLabels:
+            nodetemplate.setValueNumberOfVersions(coordinates, -1, nodeValueLabel, 1)
+        # create nodes
+        for n in range(nodesCount):
+            node = nodesetGroup.createNode(n + 1, nodetemplate)
+            cache.setNode(node)
+            for v in range(nodeValueLabelsCount):
+                coordinates.setNodeParameters(cache, -1, nodeValueLabels[v], 1, nodeValues[n][v])
+        # serialise to string
+        sir = region.createStreaminformationRegion()
+        srm = sir.createStreamresourceMemory()
+        sir.setResourceGroupName(srm, groupName)
+        region.write(sir)
+        result, exString = srm.getBuffer()
+        #print('\n',nodeValues)
+        #print('exnodeStringFromNodeValues', result, exString)
```

Comment:
Commented out debugging code should just be removed.

Reply:
Will fix.